www/pfSense-pkg-vHosts/files/usr/local/pkg/vhosts.inc

```diff
@@ -0,0 +1,367 @@
+			'@ssl_conf'      => $ssl_conf,
+			'@root_folder'   => $root_folder,
+			'@custom_config' => $custom_config
+			]);
+	}
+
+	/*--------------------*/
+	/* Build config file. */
+	/*--------------------*/
+	file_put_contents($conf_file_path, strtr($vhosts_g['conf'], [ '@servers' => $servers ]));
+	conf_mount_ro();
+}
+
+/*******************************************************************************
+* vhosts_dirty */
+/**
+* Sets/return vhosts subsystem dirty flag.
+*
+* @param  dirty  True/false to set/clear dirty flag, undef for no change.
+* @return		Returns new or current value of subsystem dirty flag.
+*
+* Config file setting: <service/startcmd>
+*******************************************************************************/
+function vhosts_dirty($dirty) {
+	global $vhosts_g;
+
+	if (isset($dirty)) {
+		if ($dirty) {
+			mark_subsystem_dirty($vhosts_g['subsystem_name']);
+		} else {
+			clear_subsystem_dirty($vhosts_g['subsystem_name']);
+		}
+	}
+
+	return is_subsystem_dirty($vhosts_g['subsystem_name']);
+}
+
+/*******************************************************************************
+* vhosts_lookup_cert */
+/**
+* Returns certificate info for certificate reference ID.
+*******************************************************************************/
+function vhosts_lookup_cert($refid) {
+	global $config;
+	$certs = $config['installedpackages']['vhosts']['cert'];
+	$i     = array_search($refid, array_column($certs, 'refid'));
+
+	/*--------------------------------------*/
+	/* FALSE will evaluate to 0. PHP sucks! */
+	/*--------------------------------------*/
+	return ($i !== FALSE ? $certs[$i] : NULL);
+}
+
+/*******************************************************************************
+* vhosts_post_install */
+/**
+* Creates rc file and builds the service config after package install.
+*
+* Config file settings: <custom_php_install_command>
+*******************************************************************************/
+function vhosts_post_install() {
+	global $vhosts_g;
+
+	write_rcfile(
+		[
+		"file"  => "vhostsd.sh",
+		"start" => "/usr/local/sbin/nginx -c '{$vhosts_g['conf_file_path']}' && rm -f '{$vhosts_g['dirty_file_path']}'",
+		"stop"  => "kill `cat {$vhosts_g['pid_file_path']}`                  && rm -f '{$vhosts_g['dirty_file_path']}'"
+		]);
+
+	vhosts_build_service_config();
+}
+
+/*******************************************************************************
+* vhosts_pre_deinstall */
+/**
+* Removes generated files before uninstall. 'pkg remove' command calls
+* '/etc/rc.packages <name> DEINSTALL' before remove and POST-DEINSTALL after.
+* <custom_php_deinstall_command> runs in POST-DEINSTALL after config file is gone.
+*
+* Config file setting: <custom_php_pre_deinstall_command>
+*******************************************************************************/
+function vhosts_pre_deinstall() {
+	global $vhosts_g;
+	exec("/bin/rm -f {$vhosts_g['conf_file_prefix']}*");
```

Comment:
As mentioned previously, test to make sure the value is sane and use [unlink_if_exists()](https://github.com/pfsense/pfsense/blob/master/src/etc/inc/util.inc#L1712) here instead of a shell exec.